install_config/aggregate_logging_sizing.adoc

```diff
@@ -0,0 +1,253 @@
+
+
+At Red Hat, we have used a simple test harness which uses http://linux.die.net/man/1/logger[logger] across the cluster nodes to make entries of different sizes at different rates in the system log. During testing simulations under a default RHEL7 installation at certain logging rates ( ~ 40 log lines per second ), we've hit the default rate limit of `rsyslogd`.
+
+*Scaling up* +
+
+As you scale up your project, the default logging environment might need some adjustments.
+After upgrading to `systemd-219-22.el7.x86_64`, we've added:
+
+----
+$IMUXSockRateLimitInterval 0
+$IMJournalRatelimitInterval 0
+----
+
+to `/etc/rsyslog.conf` and:
+
+----
+# Disable rate limiting
+RateLimitInterval=1s
+RateLimitBurst=10000
+Storage=volatile
+Compress=no
+MaxRetentionSec=5s
+----
+
+to `/etc/systemd/journald.conf`. 
+
+Now restart the services.
+
+----
+$ systemctl restart systemd-journald.service
+$ systemctl restart rsyslog.service
+----
+
+These settings accommodate for the 'bursty' nature of uploading in bulk. 
+After removing the rate limit, you may see increased CPU utilization on the system logging daemons as it processes any messages that would have previously been throttled.
+
+
+Rsyslog is configured (see ratelimit.interval, ratelimit.burst) to rate-limit entries read from the journal at 10,000 messages in 300 seconds. A good rule of thumb is to ensure that the rsyslog rate-limits always accommodate to the systemd-journald rate-limits.
+
+== Scaling up EFK logging 
+
+If you don't indicate the desired scale at first deployment using `ES_CLUSTER_SIZE`, the least disruptive way of adjusting your cluster is by re-running the deployer with the updated `ES_CLUSTER_SIZE` value and using the `MODE=reinstall` template parameter. Refer to the developers https://github.com/openshift/openshift-docs/blob/master/install_config/aggregate_logging.adoc#aggregate-logging-performing-elasticsearch-maintenance-operations[documentation] for more in depth information about this topic.
+
+----
+$ oc edit configmap logging-deployer
+  [change es-cluster-size value to 5]
+
+$ oc new-app logging-deployer-template --param MODE=reinstall
+----
+
+== Storage Considerations 
+An Elasticsearch index is a collection of shards and it's correspondent replica shards. This is how ES implements high availability internally, therefore there is little need of using hardware based mirroring RAID variants. RAID 0 can still be used to increase overall disk performance. +
+Every search request needs to hit a copy of every shard in the index.
+Each ES instance requires it's own individual storage, but an {product-title} deployment can only provide volumes shared by all of its pods, which again means that Elastic shouldn't be implemented with a single master. +
```

Comment:
- Corrected.
- I meant what you wrote in parenthesis. That there should be other master eligible nodes in quorum. I'll change single master to single node.